src/Bus/MongoBatchRepository.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace MongoDB\Laravel\Bus;
+
+use Carbon\CarbonImmutable;
+use Closure;
+use DateTimeInterface;
+use Illuminate\Bus\Batch;
+use Illuminate\Bus\BatchFactory;
+use Illuminate\Bus\DatabaseBatchRepository;
+use Illuminate\Bus\PendingBatch;
+use Illuminate\Bus\PrunableBatchRepository;
+use Illuminate\Bus\UpdatedBatchJobCounts;
+use Illuminate\Support\Carbon;
+use MongoDB\BSON\ObjectId;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\Driver\ReadPreference;
+use MongoDB\Laravel\Collection;
+use MongoDB\Laravel\Connection;
+use MongoDB\Operation\FindOneAndUpdate;
+use Override;
+
+use function date_default_timezone_get;
+use function is_string;
+use function serialize;
+use function unserialize;
+
+// Extending DatabaseBatchRepository is necessary so methods pruneUnfinished and pruneCancelled
+// are called by PruneBatchesCommand
+class MongoBatchRepository extends DatabaseBatchRepository implements PrunableBatchRepository
+{
+    private Collection $collection;
+
+    public function __construct(
+        BatchFactory $factory,
+        Connection $connection,
+        string $collection,
+    ) {
+        $this->collection = $connection->getCollection($collection);
+
+        parent::__construct($factory, $connection, $collection);
+    }
+
+    #[Override]
+    public function get($limit = 50, $before = null): array
+    {
+        if (is_string($before)) {
+            $before = new ObjectId($before);
```

Comment:
Beware that this may throw an exception if the string is not a well-formed `ObjectId`. Not sure if that's something we want to consider, but I figured I'd point it out.

Reply:
In the `store` method, the `_id` is not set, so only `ObjectId` can be inserted.

Reply:
I wanted to point this out as theoretically users could call the method with any string. No objection to assuming valid ObjectIDs are passed.